pinot-core/src/main/java/org/apache/pinot/core/query/pruner/ColumnValueSegmentPruner.java

```diff
@@ -277,7 +286,8 @@ private boolean pruneEqPredicate(IndexSegment segment, EqPredicate eqPredicate,
     // Check bloom filter
     BloomFilterReader bloomFilter = dataSource.getBloomFilter();
     if (bloomFilter != null) {
-      if (!bloomFilter.mightContain(value.toString())) {
+      Hash128 hash = hashes.computeIfAbsent(value, Hash128::new);
```

Comment:
I don't think we can use identity map here. Notice that the value might go through some conversion, so each time we are getting the same reference. We should probably just use a hash map from `String` to `Hash128`.